Runtime/CourseController/CourseControllerSetup.cs

```diff
@@ -138,5 +138,15 @@ public static void SetEnforcedCourseController(ICourseController courseControlle
         {
             enforcedCourseController = courseController;
         }
+
+        /// <summary>
+        /// Resets the <cref name="courseControllerQualifiedName"/> to the name of the course controller with the highest priority.
+        /// </summary>
+        /// <remarks>This may be used when instantiating a course controller prefab to make sure the default course controller is used.</remarks>
+        public void ResetToDefault()
+        {
+            Type courseControllerType = RetrieveDefaultControllerType();
+            courseControllerQualifiedName = courseControllerType.Name;
+        }
```

Comment:
What about doing this internally in the [Reset](https://docs.unity3d.com/ScriptReference/MonoBehaviour.Reset.html) method?

Reply:
You cannot call "Reset", I thought about that too :/ Or do you have a way to make this happen?

Reply:
Could remove and add the script again but that also isn't really nice

Reply:
True 🤔
Maybe calling it with reflection? Or making it internal?

Consider this as a suggestion, nothing that blocks merging this, I just wonder a way not to expose unnecessary APIs.

Reply:
I tried it with reflection and it technically works but makes things very complicated with the editor script and probably will lead to some bigger changes. so i would keep it like this